src/main/java/bio/terra/workspace/service/workspace/WorkspaceService.java

```diff
@@ -48,29 +50,37 @@ public WorkspaceService(
 
   /** Create a workspace with the specified parameters. Returns workspaceID of the new workspace. */
   @Traced
-  public UUID createWorkspace(
-      UUID workspaceId,
-      Optional<SpendProfileId> spendProfileId,
-      WorkspaceStage workspaceStage,
-      AuthenticatedUserRequest userReq) {
+  public UUID createWorkspace(WorkspaceRequest internalRequest, AuthenticatedUserRequest userReq) {
 
-    String description = "Create workspace " + workspaceId.toString();
+    String description = "Create workspace " + internalRequest.workspaceId().toString();
     JobBuilder createJob =
         jobService
             .newJob(
                 description,
-                UUID.randomUUID().toString(), // JobId does not need persistence for sync calls.
+                internalRequest.operationId(),
                 WorkspaceCreateFlight.class,
                 null,
                 userReq)
-            .addParameter(WorkspaceFlightMapKeys.WORKSPACE_ID, workspaceId);
-    if (spendProfileId.isPresent()) {
-      createJob.addParameter(WorkspaceFlightMapKeys.SPEND_PROFILE_ID, spendProfileId.get().id());
+            .addParameter(WorkspaceFlightMapKeys.WORKSPACE_ID, internalRequest.workspaceId());
+    if (internalRequest.spendProfileId().isPresent()) {
+      createJob.addParameter(
+          WorkspaceFlightMapKeys.SPEND_PROFILE_ID, internalRequest.spendProfileId().get().id());
     }
 
-    createJob.addParameter(WorkspaceFlightMapKeys.WORKSPACE_STAGE, workspaceStage);
-
-    return createJob.submitAndWait(UUID.class);
+    createJob.addParameter(
+        WorkspaceFlightMapKeys.WORKSPACE_STAGE, internalRequest.workspaceStage());
+
+    try {
+      return createJob.submitAndWait(UUID.class);
+    } catch (DuplicateFlightIdSubmittedException ex) {
+      // Indicates another request with the same operation ID already exists. Rather than running
+      // multiple concurrent create flights, all flights after the first will wait for and return
+      // the same result.
+      jobService.waitForJob(internalRequest.operationId());
+      return jobService
+          .retrieveJobResult(internalRequest.operationId(), UUID.class, userReq)
+          .getResult();
```

Comment:
should this be encapsulated by `JobService`? we could make it a part of submitAndWait, or another function that does this whole try catch. (Encapsulating would fix some of the leakage mentioned above.)

Reply:
moved into submit functions per below